qc-runner/src/doubletScores.r

```diff
@@ -84,6 +84,17 @@ task <- function(seurat_obj, config, task_name, sample_id){
 
     plot1_data <- lapply(unname(sample_subset$doublet_scores),function(x) {c("doubletP"=x)})
 
+
+    # Downsample plotData
+    # Handle when the number of remaining cells is less than the number of cells to downsample
+    num_cells_to_downsample <- min(max(percent_downsample / 100 * ncol(sample_subset), num_cells_to_downsample), ncol(sample_subset))
```

Comment:
What do you think about using a function for all these repeated code?

Reply:
sounds good. Refactor it out in utils?

Reply:
Refactored